src/main.rs

```diff
@@ -76,21 +78,28 @@ fn main() {
         process::exit(0);
     }
 
-    if args.flag_file.is_none() {
+
+    if args.flag_string.is_none() && args.flag_file.is_none() {
         println!("{}", USAGE);
         process::exit(0);
     }
 
-    let file = match read_file(&args.flag_file.unwrap()) {
-        Ok(file) => file,
-        Err(err) => {
-            println!("ERR Couldn't read file: {:?}", err);
-            process::exit(1)
-        }
+
+    let input = match args.flag_string.is_none() {
+        true => {
```

Comment:
Should rather be:

```rust
let input = match.args.flag_string {
    Some(source) => source + "\n", // is the extra newline necessary?
                                   // It shouldn't be a problem, but it will cause a reallocation
    None         => { /* none branch */ }
};